libibumad/umad.c

```diff
@@ -1123,3 +1151,44 @@ void umad_dump(void *umad)
 	       mad->agent_id, mad->status, mad->timeout_ms);
 	umad_addr_dump(&mad->addr);
 }
+
+int umad_get_ca_namelist(char **cas)
+{
+	struct dirent **namelist;
+	int n, i, j = 0;
+
+	n = scandir(SYS_INFINIBAND, &namelist, NULL, alphasort);
```

Comment:
Please don't add new users of sysfs, new code should use the netlink flow to get device names.

Reply:
I agree with you and posted to the ML that the best thing will be to use netlink, however, all libibumad is not converted and the right thing will be to convert all library in one shot without doing half work.